src/main/java/com/datatorrent/netlet/DefaultEventLoop.java

```diff
@@ -344,6 +348,40 @@ protected final void handleSelectedKey(final SelectionKey sk) throws IOException
 
   }
 
+  private void handleFullTasksCircularBuffer(Runnable r, int sleepMillis)
+  {
+    final Thread currentThread = Thread.currentThread();
+    if (eventThread == currentThread) {
+      int size = tasks.size();
+      if (size == tasks.capacity()) {
+        try {
+          while (size > 0) {
+            tasks.pollUnsafe().run();
+            size--;
+          }
+        } catch (RuntimeException e) {
+          logger.error("", e);
+          throw e;
+        } catch (Exception e) {
+          logger.error("", e);
```

Comment:
The EventLoop will log the exception, if nobody catches it right?


From: Vlad Rozov <notifications@github.com<mailto:notifications@github.com>>
Reply-To: DataTorrent/Netlet <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Wednesday, March 30, 2016 at 6:59 PM
To: DataTorrent/Netlet <Netlet@noreply.github.com<mailto:Netlet@noreply.github.com>>
Cc: Chetan Narsude <chetan@apache.org<mailto:chetan@apache.org>>
Subject: Re: [DataTorrent/Netlet] Better handling for the case when tasks circular buffer becomes full (#43)


In src/main/java/com/datatorrent/netlet/DefaultEventLoop.java<https://github.com/DataTorrent/Netlet/pull/43#discussion_r57992260>:

> +  private void handleFullTasksCircularBuffer(Runnable r, int sleepMillis)
> +  {
> +    final Thread currentThread = Thread.currentThread();
> +    if (eventThread == currentThread) {
> +      int size = tasks.size();
> +      if (size == tasks.capacity()) {
> +        try {
> +          while (size > 0) {
> +            tasks.pollUnsafe().run();
> +            size--;
> +          }
> +        } catch (RuntimeException e) {
> +          logger.error("", e);
> +          throw e;
> +        } catch (Exception e) {
> +          logger.error("", e);


The exception is thrown to terminate DefaultEventLoop thread. It is thrown on the DefaultEventLoop by running a task. As task does not have a way to determine what listener submitted the task, the exception can not be handled.

-
You are receiving this because you were assigned.
Reply to this email directly or view it on GitHub<https://github.com/DataTorrent/Netlet/pull/43/files/0eafb5aca8a285aa95042ce0d5edfbbda2d4ef8e#r57992260>